Set timeouts on all requests library calls

jennifer-richards opened the pull request:
Addresses [ticket 3498](https://trac.ietf.org/trac/ietfdb/ticket/3498).

  * Add DEFAULT_REQUESTS_TIMEOUT settings
  * Add timeout argument to all requests.get/requests.post calls without them
    - switch "timeout=20" arg to "timeout=settings.DEFAULT_REQUESTS_TIMEOUT"
  * Fix timeout settings in ietf/bin/ scripts
    - remove ineffectual socket.setdefaulttimeout() calls
    - use 30 seconds instead of DEFAULT_REQUESTS_TIMEOUT to preserve old assumption

meadmaker commented on ietf/doc/views_review.py:
```diff
@@ -621,7 +621,7 @@ def clean_review_url(self):
         #scheme, netloc, path, parameters, query, fragment = urlparse(url)
         if url:
-            r = requests.get(url)
+            r = requests.get(url, timeout=settings.DEFAULT_REQUESTS_TIMEOUT)
             if r.status_code != 200:
                 raise forms.ValidationError("Trying to retrieve the URL resulted in status code %s: %s.  Please provide an URL that can be retrieved." % (r.status_code, r.reason))
```
The addition of the timeout opens the possibility of raising a `requests.exceptions.Timeout` exception.  It would be better to handle that.

meadmaker commented on ietf/bin/iana-protocols-updates:
```diff
@@ -31,8 +31,11 @@ syslog.syslog("Updating history log with new RFC entries from IANA protocols pag
 
-text = requests.get(settings.IANA_SYNC_PROTOCOLS_URL).text
-rfc_numbers = parse_protocol_page(text)
+response = requests.get(
+    settings.IANA_SYNC_PROTOCOLS_URL,
+    timeout=30,
+)
```
The addition of the timeout opens the possibility of raising a requests.exceptions.Timeout exception. It would be nice to handle that, though it would be understandable if you decide that not erroring out of the command line utility is a correct behavior.

meadmaker commented on ietf/bin/rfc-editor-index-updates:
```diff
@@ -48,11 +46,18 @@ if options.skip_date:
-socket.setdefaulttimeout(30)
-rfc_index_xml = requests.get(settings.RFC_EDITOR_INDEX_URL).text
+response = requests.get(
+    settings.RFC_EDITOR_INDEX_URL,
+    timeout=30,  # seconds
+)
+rfc_index_xml = response.text
 index_data = ietf.sync.rfceditor.parse_index(io.StringIO(rfc_index_xml))
 
-errata_data = requests.get(settings.RFC_EDITOR_ERRATA_JSON_URL).json()
+response = requests.get(
+    settings.RFC_EDITOR_ERRATA_JSON_URL,
+    timeout=30,  # seconds
+)
+errata_data = response.json()
```
The addition of the timeout opens the possibility of raising a requests.exceptions.Timeout exception. It would be nice to handle that, though it would be understandable if you decide that not erroring out of the command line utility is a correct behavior.

meadmaker commented on ietf/bin/rfc-editor-queue-updates:
```diff
@@ -21,9 +20,11 @@ from ietf.utils.log import log
-socket.setdefaulttimeout(30)
-response = requests.get(settings.RFC_EDITOR_QUEUE_URL).text
-drafts, warnings = parse_queue(io.StringIO(response))
+response = requests.get(
+    settings.RFC_EDITOR_QUEUE_URL,
+    timeout=30,  # seconds
+)
```
The addition of the timeout opens the possibility of raising a requests.exceptions.Timeout exception. It would be nice to handle that, though it would be understandable if you decide that not erroring out of the command line utility is a correct behavior.

meadmaker commented on ietf/meeting/utils.py:
```diff
@@ -123,7 +123,7 @@ def create_proceedings_templates(meeting):
     try:
-        attendees = requests.get(url).json()
+        attendees = requests.get(url, timeout=settings.DEFAULT_REQUESTS_TIMEOUT).json()
     except (ValueError, HTTPError):
```
Add the `requests.exceptions.Timeout` to the handled exceptions.

meadmaker commented on ietf/stats/utils.py:
```diff
@@ -230,7 +230,10 @@ def get_meeting_registration_data(meeting):
     num_created = 0
     num_processed = 0
-    response = requests.get(settings.STATS_REGISTRATION_ATTENDEES_JSON_URL.format(number=meeting.number))
+    response = requests.get(
+        settings.STATS_REGISTRATION_ATTENDEES_JSON_URL.format(number=meeting.number),
+        timeout=settings.DEFAULT_REQUESTS_TIMEOUT,
+    )
```
The addition of the timeout opens the possibility of raising a `requests.exceptions.Timeout` exception. While this code is currently only called from a `manage.py` task, it might be used elsewhere at a later point.  Please handle the exception.

meadmaker commented on ietf/sync/iana.py:
```diff
@@ -69,7 +69,7 @@ def fetch_changes_json(url, start, end):
     password = settings.IANA_SYNC_PASSWORD
     headers = { "Authorization": "Basic %s" % force_str(base64.encodebytes(smart_bytes("%s:%s" % (username, password)))).replace("\n", "") }
-    text = requests.get(url, headers=headers).text
+    text = requests.get(url, headers=headers, timeout=settings.DEFAULT_REQUESTS_TIMEOUT).text
```
The addition of the timeout opens the possibility of raising a requests.exceptions.Timeout exception. It would be nice to handle that, though it would be understandable if you decide that erroring out of the command line utility is a correct behavior.

meadmaker commented:
Also, Robert had asked:

> Please emit something into syslog when a timeout occurs at each of these places?

I don't see that in any of the code here.

jennifer-richards commented:
I think I've addressed everything. I've tried to handle the exceptions consistently with the nearby code, which varied between places.

I also added a `from exc` to the exceptions raised in `hedgedoc.py`. This should give a more complete backtrace in the debugger if it comes up.

The timeouts weren't logged because I created the PR before Robert made that request. I think they all result in a log entry now.

meadmaker reviewed (approved):
Looks great
